p2p/discovery/refresher.go

```diff
@@ -62,28 +61,33 @@ func (r *refresher) Bootstrap(ctx context.Context, minPeers int) error {
 	// if there were any results but we didn't reach minPeers it will try the same procedure
 	// using the results as servers to query. every failed loop will wait a backoff
 	// to let other nodes populate before flooding with queries.
-
+	seed := false
+	size := r.book.NumAddresses()
+	if size == 0 {
+		r.book.AddAddresses(r.bootNodes, r.localAddress)
+		servers = r.bootNodes
+		seed = true
+	} else {
+		srv := r.book.AddressCache()
+		servers = srv[:common.Min(minPeers, len(srv))]
+	}
 loop:
 	for {
-		size := r.book.NumAddresses()
-		if size == 0 {
-			r.book.AddAddresses(r.bootNodes, r.book.localAddress)
-			servers = r.bootNodes
-		}
-
-		if size < minPeers+len(r.bootNodes) {
-			r.logger.Info("Bootstrap: starting with %v sized table", size)
-			res := r.requestAddresses(servers)
-			if len(res) > 0 {
-				servers = res
-			}
-			tries++
-			r.logger.Info("Bootstrap : %d try gave %v results", tries, len(res))
+		r.logger.Info("Bootstrap: starting with %v sized table", size)
+		res := r.requestAddresses(servers)
+		if len(res) > 0 {
+			servers = res
```

Comment:
don't you want to limit the number of servers used?

Reply:
But in case we get many peers from the bootstrap and we ask all of them for their peers we might be bombarded with replies. Isn't it a concern?

Reply:
We're sending 3 queries at a time so we'll not be bombarded. though if we did get "many" peers we'll won't query much further since we got our required amount.

Reply:
Sounds good, I forgot that we are limiting the number of concurrent requests.